ietf/draft-ietf-jmap-calendars-18.txt

```diff
@@ -0,0 +1,3360 @@
+   to the following:
+
+   *  Text should be matched in a case-insensitive manner.
+   *  Text contained in either (but matched) single or double quotes
+      should be treated as a *phrase search*, that is a match is
+      required for that exact sequence of words, excluding the
+      surrounding quotation marks.  Use \", \' and \\ to match a literal
+      ", ' and \ respectively in a phrase.
+   *  Outside of a phrase, white-space should be treated as dividing
+      separate tokens that may be searched for separately in the event,
+      but must all be present for the event to match the filter.
+   *  Tokens may be matched on a whole-word basis using stemming (so for
+      example a text search for bus would match "buses" but not
+      "business").
+
+5.10.2.  Sorting
+
+   The following properties MUST be supported for sorting:
+
+   *  start
+   *  uid
+   *  recurrenceId
+
+
+
+
+Jenkins & Douglass       Expires 9 December 2024               [Page 34]
+
+Internet-Draft               JMAP Calendars                    June 2024
+
+
+   The following properties SHOULD be supported for sorting:
+
+   *  created
+   *  updated
+
+5.11.  CalendarEvent/queryChanges
+
+   This is a standard "/queryChanges" method as described in Section 5.6
+   of [RFC8620].
+
+5.12.  CalendarEvent/parse
+
+   This method allows the client to parse blobs as iCalendar files
+   [RFC5545] to get CalendarEvent objects.  This can be used to parse,
+   display, and import information from iCalendar files without having
+   to implement iCalendar parsing in the client.  Server support for
+   this is optional, and indicated via the
+   urn:ietf:params:jmap:calendars:parse capability, as per
+   Section 1.5.3.
+
+   The following metadata properties on the CalendarEvent objects will
+   be null if requested:
+
+   *  id
+   *  baseEventId
+   *  calendarIds
+   *  isDraft
+   *  isOrigin
+
+   The "CalendarEvent/parse" method takes the following arguments:
+
+   *  *accountId*: Id
+      The id of the account to use.
+   *  *blobIds*: Id[]
+      The ids of the blobs to parse.
+   *  *properties*: String[]
+      If supplied, only the properties listed in the array are returned
+      for each CalendarEvent object.  If omitted, defaults to all the
+      properties.
+
+   The response object contains the following arguments:
+
+   *  *accountId*: Id
+      The id of the account used for the call.
+   *  *parsed*: Id[CalendarEvent[]]|null
+      A map of blob ids to parsed CalendarEvent objects representations
+      for each successfully parsed blob, or null if none.
+   *  *notFound*: Id[]|null
+
+
+
+Jenkins & Douglass       Expires 9 December 2024               [Page 35]
+
+Internet-Draft               JMAP Calendars                    June 2024
+
+
+      A list of blob ids given that could not be found, or null if none.
+   *  *notParsable*: Id[]|null
+      A list of blob ids given that corresponded to blobs that could not
```

Comment:
There's no standardised categories of error we could return here I think, and the string gets into localisation issues. While I agree it could be useful, the current form is consistent with the `Email/parse` method which we've already standardised, so I'm inclined to keep it like this.